### PR TITLE
fix: [IAC-1288]: Fix evaluation status

### PR DESCRIPTION
**Pull request by willd0yle**
### Summary
Fix evaluation status when governance status is an empty string. A scenario occurs where there is governance data however the status has not been set yet.

This fix sets the status to Error if either iacm or governance has an error, warning if either has a warning otherwise it will be set to pass until the data has been updated.

#### Screenshots

NA

#### PR Checklist

<details>
<summary>Please check if your PR fulfils the following requirements</summary>

- [ ] Tests for the changes have been added. Ideally, include a test that fails without this PR but passes with it.
- [ ] Docs have been [added/updated](https://harness.atlassian.net/jira/software/c/projects/DOC/boards/40).
</details>

<details>
<summary>Use the following comments to re-trigger PR Checks</summary>

- Jest: `retrigger jest`
- Prettier: `retrigger prettier`
- Type Check: `retrigger typecheck`
- ESLint: `retrigger eslint`
- Sonar: `retrigger sonar`
- Standards: `retrigger standards`
- Build: `retrigger build`
- Title Check: `retrigger titlecheck`
- Feature Name Check: `trigger featurenamecheck`
- Coverage: `retrigger coverage`
- Rebase: `trigger rebase`
- Cypress Rest: `retrigger cypress-rest`
- Cypress CD: `retrigger cypress-cd`
- Cypress Pipeline: `retrigger cypress-pipeline`
- Cypress CV: `retrigger cypress-cv`
- Fix Prettier: `fix prettier`
- MessageMetadata: `trigger messagecheck`
- GitLeaks: `trigger gitLeaks`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

### Rebase Instructions:
To rebase your branch onto the latest changes in the target branch, simply use the following command in a comment: `/rebase`

### CODE REVIEW

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 1]
```diff
@@ -1,29 +1,67 @@
...
+  const queryParams = {
+    accountIdentifier: accountId,
+    orgIdentifier,
+    projectIdentifier,
+    includeChildScopes: true,
+    module: 'iacm',
+    entity: `iacm-${executionIdentifier}`,
+    lazy: true
+  }
+  const { data, loading, error, refetch } = useGetEvaluationList({ queryParams })
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:36]: I think the `lazy` is in the wrong place.
```suggestion
  }
  const { data, loading, error, refetch } = useGetEvaluationList({
    queryParams,
    lazy: true
  })
```

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx [hunk 2]
```diff
@@ -1,29 +1,67 @@
...
+import { useFeatureFlag } from '@modules/10-common/hooks/useFeatureFlag'
+import { Evaluation, useGetEvaluationList } from 'services/pm'
+import { formatMetaData } from './utils'
 
 export default function ExecutionPolicyEvaluationsView(): React.ReactElement | null {
-  const { accountId, module } = useParams<PipelineType<ExecutionPathProps>>()
+  const { showError } = useToaster()
+  const IACM_ENABLED = useFeatureFlag(FeatureFlag.IACM_ENABLED)
+  const { accountId, orgIdentifier, projectIdentifier, executionIdentifier, module } =
+    useParams<PipelineType<ExecutionPathProps>>()
```
Comment (knagurski) [inline src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:24]: I always just set this as `Record<string, string>` coz I couldn't be arsed trying to come up with the right type when it doesn't guarantee things exist anyway. No need to change, just a mini-rant

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx (hunk 1 above)
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx]: done

File: src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx (hunk 2 above)
Comment (willd0yle) [reply src/modules/70-pipeline/pages/execution/ExecutionPolicyEvaluationsView/ExecutionPolicyEvaluationsView.tsx:24]: Preach

**Review by knagurski — DISMISSED** [review]
Seems inoffensive

**Review by rameshx — APPROVED** [review]

**Pull request merged**